lib/multi-write.ts

```diff
@@ -133,12 +137,12 @@ export async function pipeSourceToDestinations(
 			size = state.size;
 			bytesWritten = progress.position;
 		}
-		if ((size !== undefined) && (bytesWritten !== undefined)) {
+		if ((size !== undefined) && (bytesWritten !== undefined) && (bytesWritten <= size)) {
 			percentage = bytesWritten / size * 100;
-			eta = (size - bytesWritten) / progress.speed;
+			eta = getEta(bytesWritten, size, progress.speed);
 		} else if ((state.rootStreamSpeed !== undefined) && (state.rootStreamPosition !== undefined) && (state.compressedSize !== undefined)) {
 			percentage = state.rootStreamPosition / state.compressedSize * 100;
-			eta = (state.compressedSize - state.rootStreamPosition) / state.rootStreamSpeed;
+			eta = getEta(state.rootStreamPosition, state.compressedSize, state.rootStreamSpeed);
 		}
```

Comment:
@zvin I guess that if none of the if/elseif conditions are satisfied, the `eta` will just be **undefined**. Is this correct/intended ?

Reply:
@thundron , correct, there might be cases where we don't know the image compressed or uncompressed size (not in Etcher though).